lib/SILGen/SILGenPoly.cpp

```diff
@@ -1360,7 +1360,7 @@ namespace {
             output = SGF.emitManagedRetain(Loc, output.getValue());
           }
 
-          if (output.getOwnershipKind() == ValueOwnershipKind::Owned) {
+          if (output.getOwnershipKind() == ValueOwnershipKind::Guaranteed) {
```

Comment:
This doesn't look right to me either. I'll clean it up. We don't support implicit borrowing via in_guaranteed do we?